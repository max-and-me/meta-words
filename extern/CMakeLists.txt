cmake_minimum_required(VERSION 3.19.0)

# Add your dependencies here, e.g. googletest
# add_subdirectory(googletest)

add_subdirectory(tiny-process-library)
add_subdirectory(fast-cpp-csv-parser)