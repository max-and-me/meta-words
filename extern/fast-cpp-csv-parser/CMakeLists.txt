cmake_minimum_required(VERSION 3.14.0)

include(FetchContent)

FetchContent_Declare(
    fast-cpp-csv-parser
    GIT_REPOSITORY https://github.com/ben-strasser/fast-cpp-csv-parser.git
    GIT_TAG 758f5665705a38da55322511d5b22fe2679918fe
)

FetchContent_MakeAvailable(fast-cpp-csv-parser)

add_library(fast-cpp-csv-parser INTERFACE)

target_include_directories(fast-cpp-csv-parser
    INTERFACE 
        ${fast-cpp-csv-parser_SOURCE_DIR}
)
