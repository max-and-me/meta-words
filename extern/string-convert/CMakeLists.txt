cmake_minimum_required(VERSION 3.14.0)

include(FetchContent)

FetchContent_Declare(
    string-convert
    GIT_REPOSITORY https://github.com/max-and-me/string-convert.git
    GIT_TAG b3b431763a0bc55ffdceadc190801b6e856788f2
)

FetchContent_MakeAvailable(string-convert)
