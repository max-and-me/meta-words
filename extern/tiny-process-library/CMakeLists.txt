cmake_minimum_required(VERSION 3.14.0)

include(FetchContent)

FetchContent_Declare(
    tiny-process-library
    GIT_REPOSITORY https://gitlab.com/eidheim/tiny-process-library.git
    GIT_TAG v2.0.4
)

option(BUILD_TESTING "" OFF)

FetchContent_MakeAvailable(tiny-process-library)
