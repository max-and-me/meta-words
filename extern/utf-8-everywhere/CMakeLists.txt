cmake_minimum_required(VERSION 3.14.0)

include(FetchContent)

FetchContent_Declare(
    utf-8-everywhere
    GIT_REPOSITORY https://github.com/max-and-me/utf-8-everywhere.git
    GIT_TAG b9dd0d89f49446d02d1a3a242aef484d625418db
)

FetchContent_MakeAvailable(utf-8-everywhere)
