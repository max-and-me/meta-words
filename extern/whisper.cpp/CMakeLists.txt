cmake_minimum_required(VERSION 3.14.0)

include(FetchContent)

FetchContent_Declare(
    whisper.cpp
    GIT_REPOSITORY https://github.com/ggerganov/whisper.cpp.git
    GIT_TAG v1.7.5
)

option(WHISPER_BUILD_EXAMPLES "whisper: build examples" ON)

FetchContent_MakeAvailable(whisper.cpp)

# Cache the models download directory
set(MAM_WHISPER_CPP_MODEL_DOWNLOAD_DIR "${PROJECT_BINARY_DIR}/whisper.cpp/models" CACHE PATH "MAM whisper.cpp models models diretory")

# Keep the source dir for later usage
# message(FATAL_ERROR "[MAM] whisper.cpp_SOURCE_DIR: ${whisper.cpp_SOURCE_DIR}")
set_target_properties(whisper
    PROPERTIES
        whisper_SOURCE_DIR ${whisper.cpp_SOURCE_DIR}
)

# Only download if not exist already
set(MAM_WHISPER_CPP_MODEL_FILE_NAME "ggml-base.en.bin")
if(EXISTS "${MAM_WHISPER_CPP_MODEL_DOWNLOAD_DIR}/${MAM_WHISPER_CPP_MODEL_FILE_NAME}")
    return()
endif()
message(STATUS "[MAM] Downloading model file ...")
file(DOWNLOAD 
    https://huggingface.co/ggerganov/whisper.cpp/resolve/main/${MAM_WHISPER_CPP_MODEL_FILE_NAME}
    ${MAM_WHISPER_CPP_MODEL_DOWNLOAD_DIR}/${MAM_WHISPER_CPP_MODEL_FILE_NAME}
    SHOW_PROGRESS
)
