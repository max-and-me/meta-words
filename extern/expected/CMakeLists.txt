cmake_minimum_required(VERSION 3.14.0)

include(FetchContent)

FetchContent_Declare(
    std-expected
    GIT_REPOSITORY https://github.com/TartanLlama/expected.git
    GIT_TAG v1.1.0
)

FetchContent_MakeAvailable(std-expected)
